File: javascript/src/models/listResponseStreamEventTypeOut.ts
// this file is @generated
import {
  type StreamEventTypeOut,
  StreamEventTypeOutSerializer,
} from "./streamEventTypeOut";

export interface ListResponseStreamEventTypeOut {
  data: StreamEventTypeOut[];
  iterator: string | null;
  prevIterator?: string | null;
  done: boolean;
}

export const ListResponseStreamEventTypeOutSerializer = {
  _fromJsonObject(object: any): ListResponseStreamEventTypeOut {
    return {
      data: object["data"].map((item: StreamEventTypeOut) =>
        StreamEventTypeOutSerializer._fromJsonObject(item)
      ),
      iterator: object["iterator"],
      prevIterator: object["prevIterator"],
      done: object["done"],
    };
  },

  _toJsonObject(self: ListResponseStreamEventTypeOut): any {
    return {
      data: self.data.map((item) => StreamEventTypeOutSerializer._toJsonObject(item)),
      iterator: self.iterator,
      prevIterator: self.prevIterator,
      done: self.done,
    };
  },
};

File: python/svix/models/list_response_stream_event_type_out.py
# this file is @generated
import typing as t

from .common import BaseModel
from .stream_event_type_out import StreamEventTypeOut


class ListResponseStreamEventTypeOut(BaseModel):
    data: t.List[StreamEventTypeOut]

    done: bool

    iterator: t.Optional[str]

    prev_iterator: t.Optional[str] = None
